proposed/event-dispatcher-meta.md

```diff
@@ -44,9 +44,36 @@ On further review, it was determined that Collection was a special case of Objec
 
 Notification can safely be done asynchronously (including delaying it through a queue) but Modification by nature involve passing data back to the caller and thus must be synchronous.  Despite that difference the Working Group determined that the use cases were close enough to be considered in a single PSR.  The two different workflows however are represented by two different interfaces, `MessageNotifierInterface` (for Messages) and `TaskProcessorInterface` (for Tasks).
 
-### 4.2 Immutable events
+### 4.2 Intended use cases
 
-Initially the Working Group wished to define all Events as immutable message objects, similar to PSR-7.  However, that proved problematic in the Processor case.  In both of those cases Listeners needed a way to return data to the caller.  In concept there were three possible avenues:
+Despite their similarities, the Message and Task pipelines have two distinct workflows and use cases.
+
+| Messages                      | Tasks
+|-------------------------------|-------------------------------
+| Must be immutable             | May me mutable
+| Must be serializable          | May be serializable
+| May be delayed                | Must be processed immediately
```

Comment:
I'm not entirely convinced that tasks must be processed immediately. As @WyriHaximus demonstrated last week, a `Task` could compose one or more promises, which the listeners either create or react to. As such, processing could be delayed in them as well.

Reply:
True; although in that case I would argue that the listener is processing the event by enqueuing the async action.

Perhaps "listeners must be called immediately" (even if the listener then does something async)?